api/src/integrations/core-documents/datasource.js

```diff
@@ -68,17 +68,17 @@ class CoreDocumentsDataSource {
       const jobCount = process.env.PARALLEL_IMPORT_LIMIT ? parseInt(process.env.PARALLEL_IMPORT_LIMIT) : 1;
 
       const permits = await this.getPermits();
-
-      this.status.itemTotal = permits.length;
+      const numPermits = permits.length;
+      this.status.itemTotal = numPermits;
       await this.taskAuditRecord.updateTaskRecord({ itemTotal: this.status.itemTotal });
 
       const permitUtils = new PermitUtils(this.auth_payload, RECORD_TYPE.Permit);
 
       // Push records to proccess into a Promise array to be processed in parallel.
-      for (let i = 0; i < permits.length; i += jobCount) {
+      for (let i = 0; i < numPermits; i += jobCount) {
```

Comment:
This could be a good opportunity to pull this code out into a function and write a test for it. This kind of code can be confusing to read. 

Not a blocker, I know we didn't write this code we are just updating the permit length reference.